include/nbl/ext/DebugDraw/builtin/hlsl/aabb_instances.vertex.hlsl

```diff
@@ -0,0 +1,35 @@
+#pragma shader_stage(vertex)
+
+#include "nbl/builtin/hlsl/math/linalg/fast_affine.hlsl"
+#include "nbl/builtin/hlsl/glsl_compat/core.hlsl"
+#include "nbl/builtin/hlsl/bda/__ptr.hlsl"
+#include "common.hlsl"
+
+using namespace nbl::hlsl;
+using namespace nbl::ext::debug_draw;
+
+[[vk::push_constant]] SPushConstants pc;
+
+[shader("vertex")]
+PSInput main()
+{
+    const float32_t3 unitAABBVertices[8] = {
+        float32_t3(0.0, 0.0, 0.0),
+        float32_t3(1.0, 0.0, 0.0),
+        float32_t3(0.0, 0.0, 1.0),
+        float32_t3(1.0, 0.0, 1.0),
+        float32_t3(0.0, 1.0, 0.0),
+        float32_t3(1.0, 1.0, 0.0),
+        float32_t3(0.0, 1.0, 1.0),
+        float32_t3(1.0, 1.0, 1.0)
+    };
+
+    PSInput output;
+    float32_t3 vertex = unitAABBVertices[glsl::gl_VertexIndex()];
```

Comment:
its better to use arithmetic for this, e.g.
```cpp
const uint32_t index = glsl::gl_VertexIndex();
cosnt float32_t vertex = (promote<uint32_t3>(index)>>uint32_t(0,2,1)) & 0x1u;
```

Also its slightly annoying that that you took Y coordinate to be more important than Z